libs/core/src/lib/checkbox/checkbox/checkbox.component.ts

```diff
@@ -208,4 +208,11 @@ export class CheckboxComponent implements ControlValueAccessor {
     private _isSpaceBarEvent(event: KeyboardEvent): boolean {
         return event.keyCode === 32;
     }
+
+    /** Method to trigger change detection in component */
+    private _detectChanges(): void {
+        if (!this._changeDetectorRef['destroyed']) {
+            this._changeDetectorRef.detectChanges();
+        }
+    }
```

Comment:
```
if (!(this.changeDetector as ViewRef).destroyed) {
  this._changeDetector.detectChanges()
}
```
If you want to make use of the types but your solution is great so I will approve